### ISSUE TITLE
Setting an invalid value in a time field while doing update_attributes! sets the value to nil

### ISSUE THREAD

**Issue by hasitha** [clemens/delocalize#78]
I noticed that setting an invalid time value causes the value to get saved as nil without any validations failing. For an instance, 

```
some_model_intance.update_attributes!(a_datetime_field: "blah")
```

succeeds with the a_datetime_field value being set to nil.

This is happening due to the way Time.zone.parse_localized(time) is handling parsing of invalid strings in the setter in active_record_rails3.rb and active_record_rails4.rb.

In this situation, shouldn't the expected behaviour be to fail and preserve the old value?

**Comment by clemens** [comment]
Thanks for raising the issue. I'm happy to merge a pull request that fixes this in the proper way.

**Comment by DexterTheDragon** [comment]
This issue was essentially fixed in #85

**Comment by clemens** [comment]
Great to know, thanks!
